### ISSUE TITLE
ast_manager LEAKED: 9 for String formula (z3str3)

### ISSUE THREAD

**Issue by rainoftime** [Z3Prover/z3#3764]
Hi, for the following formula
~~~~
(set-option :smt.string_solver z3str3)
(declare-const Str4 String)
(declare-const Str11 String)
(declare-const Str16 String)
(assert (str.contains Str16 (str.++ Str11 "" Str4 Str16 "")))
(check-sat)
~~~~

z3 (commit 080dbb1) throws a warning
~~~~
ast_manager LEAKED: 9
Leaked: decl Str4 :: String[bv[8]]
decl Str11 :: String[bv[8]]
(= Str11 Str4)
id: 214
~~~~

**Comment by rainoftime** [comment]
There is no info. like "warning:....",   and the leak does not occur upon timeout.
So I report the issue.

**Comment by rainoftime** [comment]
Fixed at 26c96e49cd6
